src/main/java/com/contrastsecurity/maven/plugin/sdkx/ContrastScanSDK.java

```diff
@@ -51,117 +22,57 @@ public ContrastScanSDK(final ContrastSDK contrast, final String restURL) {
+   * Retrieve a scan by its ID
    *
-   * <p>Visible for testing
+   * @param organizationId unique ID for the user's organization
+   * @param projectId unique ID of the Scan project to which the scan belongs
+   * @param scanId unique ID of the scan
+   * @return {@code Scan} record
+   * @throws IOException when an IO error occurs while uploading the file
+   * @throws UnauthorizedException when Contrast rejects this request as unauthorized
    */
-  static String determineMime(final Path file) throws IOException {
-    // trust the content type Java can determine
-    final String contentType = Files.probeContentType(file);
-    if (contentType != null) {
-      return contentType;
-    }
-    // special checks for Java archive types, because not all of these types are identified by
-    // Files.probeContentType(file) and we want to make sure we handle Java extensions correctly
-    // since users of this code will most likely be uploading Java artifacts
-    final String name = file.getFileName().toString();
-    if (name.endsWith(".jar") || name.endsWith(".war") || name.endsWith(".ear")) {
-      return "application/java-archive";
-    }
-    return "application/octet-stream";
-  }
+  Scan getScanById(String organizationId, String projectId, String scanId)
+      throws IOException, UnauthorizedException;
 
   /**
-   * Starts a new scan
+   * Retrieves scan results in <a href="https://sarifweb.azurewebsites.net">SARIF</a>
    *
    * @param organizationId unique ID for the user's organization
-   * @param request parameters for requesting a new scan
-   * @return new scan operation
+   * @param projectId unique ID of the Scan project to which the scan belongs
+   * @param scanId unique ID of the scan
+   * @return {@code InputStream} for reading the SARIF response from Contrast
    * @throws IOException when an IO error occurs while uploading the file
    * @throws UnauthorizedException when Contrast rejects this request as unauthorized
    */
-  public Scan startScan(final String organizationId, final StartScanRequest request)
-      throws IOException, UnauthorizedException {
-    // requests made with ContrastSDK.makeRequest must have their path prepended with "/"
-    final String path =
-        String.join(
-            "/",
-            "",
-            "sast",
-            "organizations",
-            organizationId,
-            "projects",
-            request.getProjectId(),
-            "scans");
-    final String json = gson.toJson(request);
-    try (Reader reader =
-        new InputStreamReader(
-            contrast.makeRequestWithBody(HttpMethod.POST, path, json, MediaType.JSON))) {
-      return gson.fromJson(reader, Scan.class);
-    }
-  }
+  InputStream getSarif(String organizationId, String projectId, String scanId)
+      throws IOException, UnauthorizedException;
 
-  private static final String CRLF = "\r\n";
+  /**
+   * Retrieves scan results in <a href="https://sarifweb.azurewebsites.net">SARIF</a>
```

Comment:
If we think that we should do `shutdown()` in the case where `waitForResults == true`, then I'd prefer to use `shutdown()` in both cases. When `waitForResults == false`, and we call `shutdown`, this happens:

1. `shutdown()` returns immediately and we cede control back to Maven to run the next plugin
2. The `ExecutorService` actually shuts down when the last task finishes. In this case, there is at most one task still running and it is the initial poll to the Scan API for the scan's status.

Do you think we should `shutdownNow()` in both cases? In the case where we wait for results, we should be done with the `ExecutorService` anyway

Reply:
I think I kinda forgot that we’re doing a `finally` here (instead of just a `catch`). I think `shutdown` is actually better in this case since it would wait until the task is done, so I’d say leave it as-is.